modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/PicklistObjectFieldBusinessType.java

```diff
@@ -112,10 +114,32 @@ public PropertyDefinition.PropertyType getPropertyType() {
 
 	@Override
 	public void predefineObjectFieldSettings(
-			ObjectField newObjectField, ObjectField oldObjectField)
+			ObjectField newObjectField, ObjectField oldObjectField,
+			List<ObjectFieldSetting> objectFieldSettings)
 		throws PortalException {
 
 		if (oldObjectField == null) {
+			for (ObjectFieldSetting objectFieldSetting : objectFieldSettings) {
+				if (!StringUtil.equals(
+						objectFieldSetting.getName(),
+						ObjectFieldSettingConstants.NAME_STATE_FLOW)) {
+
+					continue;
+				}
+
+				if (objectFieldSetting.getObjectStateFlow() != null) {
+					ObjectStateFlow objectStateFlow =
+						objectFieldSetting.getObjectStateFlow();
+
+					_objectStateFlowLocalService.addObjectStateFlow(
+						newObjectField.getUserId(),
+						newObjectField.getObjectFieldId(),
+						objectStateFlow.getObjectStates());
+
+					return;
```

Comment:
@murilostodolni we don't need to `addDefaultObjectStateFlow` in this case, right?